examples/vmseries_scaleset/main.tf

```
@@ -14,119 +44,240 @@ resource "random_password" "this" {
   override_special = "_%@"
 }
 
-# Create the network required for the topology.
+# Create the transit network which holds the VM-Series
+# (both inbound and outbound ones).
 module "vnet" {
   source = "../../modules/vnet"
 
   virtual_network_name    = var.virtual_network_name
-  resource_group_name     = azurerm_resource_group.this.name
+  create_virtual_network  = var.create_virtual_network
+  resource_group_name     = local.inbound_resource_group.name
   location                = var.location
   address_space           = var.address_space
   network_security_groups = var.network_security_groups
   route_tables            = var.route_tables
   subnets                 = var.subnets
+  tags                    = merge(var.tags, var.panorama_tags, var.vnet_tags)
+}
+
+# Allow access from outside to Management interfaces of VM-Series.
+resource "azurerm_network_security_rule" "mgmt" {
+  name                        = "vmseries-mgmt-allow-inbound"
+  resource_group_name         = local.inbound_resource_group.name
+  network_security_group_name = "sg_mgmt"
+  access                      = "Allow"
+  direction                   = "Inbound"
+  priority                    = 1000
+  protocol                    = "*"
+  source_port_range           = "*"
+  source_address_prefixes     = var.allow_inbound_mgmt_ips
+  destination_address_prefix  = "*"
+  destination_port_range      = "*"
+
+  depends_on = [module.vnet]
 }
 
 ### LOAD BALANCERS ###
-# Create the inbound load balancer
+# Create the inbound load balancer.
 module "inbound_lb" {
   source = "../../modules/loadbalancer"
 
-  name                = var.lb_public_name
-  resource_group_name = azurerm_resource_group.this.name
-  location            = var.location
-  frontend_ips        = var.public_frontend_ips
+  name                              = var.lb_public_name
+  resource_group_name               = local.inbound_resource_group.name
+  location                          = var.location
+  frontend_ips                      = var.public_frontend_ips
+  enable_zones                      = var.enable_zones
+  tags                              = merge(var.tags, var.panorama_tags)
+  network_security_group_name       = "sg_pub_inbound"
+  network_security_allow_source_ips = coalescelist(var.allow_inbound_data_ips, var.allow_inbound_mgmt_ips)
 }
 
-locals {
-  private_frontend_ips = { for k, v in var.private_frontend_ips : k => {
-    subnet_id                     = module.vnet.subnet_ids["private"]
-    private_ip_address_allocation = try(v.private_ip_address_allocation, null)
-    private_ip_address            = var.olb_private_ip
-    rules                         = v.rules
+# Create the outbound load balancer.
+module "outbound_lb" {
+  source = "../../modules/loadbalancer"
+
+  name                = var.lb_private_name
```

Nit:
```suggestion
  name                = var.outbound_lb_name
```
or
```suggestion
  name                = var.olb_name
```
To be compliant to the `olv_private_ip`.